EchoPro/tests/test_strata_mean_sigma_bs.py

```diff
@@ -0,0 +1,136 @@
+        }
+    )
+
+    ### Dummy parameters
+    objS.config[ 'TS_length_regression_parameters' ][ 'pacific_hake' ] = { 'species_code': 19350 ,
+                                                                           'TS_L_slope': 20.0 ,
+                                                                           'TS_L_intercept': -84.1 ,
+                                                                           'length_units': 'cm' }
+    
+    ### Define dummy `species_id` code
+    species_id = 19350
+
+    #----------------------------------
+    ### Run tests: `strata_mean_sigma_bs`
+    #----------------------------------
+    ### Evaluate whether non-specified `species_id` code are removed 
+    # `specimen_df`: 20 rows -> 19 rows
+    specimen_df_copy = objS.biology[ 'specimen_df' ].copy( )
+    specimen_df_copy = specimen_df_copy[ specimen_df_copy.species_id == species_id ]
+    assert specimen_df_copy.shape[ 0 ] == 19
+
+    # `length_df`: 20 rows -> 19 rows
+    length_df_copy = objS.biology[ 'length_df' ].copy( )
+    length_df_copy = length_df_copy[ length_df_copy.species_id == species_id ]
+    assert length_df_copy.shape[ 0 ] == 19
+
+    #----------------------------------
+    ### Next step: Concatenate the two dataframes
+    # Re-bin `specimen_df_copy`
+    spec_df_reframed = (
+        specimen_df_copy
+        .groupby( [ 'haul_num' , 'stratum_num' , 'species_id', 'length' ] )
+        .apply( lambda x: len( x[ 'length' ] ) )
+        .reset_index( name = 'length_count' )
+    )
+
+    # Concatenate
+    all_length_df = pd.concat( [ spec_df_reframed , length_df_copy ] , 
+                               join = 'inner' )
+    assert all_length_df.shape[ 0 ] == 38
+
+    #----------------------------------
+    ### TS-length parameterization & modeling
+    # Fit parameters
+    ts_length_parameters = objS.config[ 'TS_length_regression_parameters' ][ 'pacific_hake' ]
+    slope = ts_length_parameters[ 'TS_L_slope' ]
+    intercept = ts_length_parameters[ 'TS_L_intercept' ]
+    assert slope == 20.0
+    assert intercept == -84.1
+
+    # Predict `TS`
+    all_length_df[ 'TS' ] = ts_length_regression( all_length_df[ 'length' ] , slope , intercept )
+    assert np.isclose( np.median( all_length_df[ 'TS' ] ) , -49.675 )
+
+    # Linearize to `sigma_bs`
+    all_length_df[ 'sigma_bs' ] = to_linear( all_length_df[ 'TS' ] )
+    np.isclose( all_length_df[ 'sigma_bs' ].mean( ) , 1.3395e-5 )
+
+    #----------------------------------
+    ### Calculate mean `sigma_bs` per `haul_num` and then `stratum_num`
+    # `haul_num`
+    mean_haul_sigma_bs = (
+        all_length_df
+        .groupby( [ 'haul_num' , 'stratum_num' , 'species_id' ] )[ [ 'sigma_bs' , 'length_count' ] ]
+        .apply(lambda x: np.average( x[ 'sigma_bs' ] , weights= x[ 'length_count' ] ) )
+        .to_frame( 'sigma_bs_mean' )
+        .reset_index( )
+    )
+    assert mean_haul_sigma_bs.shape[ 0 ] == 14
```

Comment:
Here I would also suggest testing for the exact values, in addition to the shape.